### DISCUSSION TITLE
Guidance for layouts

### DISCUSSION THREAD

**Discussion by darthmall** [zellij-org/zellij#2162] [General]
I'm curious how folks are making use of layouts and if there are any recommended courses of action or places where people share their configs.

Do you just build <i>ad hoc</i> layouts with the `add-pane` command? Do you have a couple of generic layouts you use for different tasks, or do you define per-project layouts that start up different command in different pane arrangements and tabs? Do you keep the layout files in the project directory (the way IDEs keep files in project directories) or do you keep them all in your LAYOUTS directory?

**Comment by imsnif** [comment]
Hey! So the layout system in its current incarnation is pretty new (it's only a few months old). I think we're just about at the stage of figuring out this sort of problem.

Personally, I keep a few default layouts which I trigger with shell aliases (eg. `e2e` opens a new tab in my current session running my end-to-end tests, `add-screen-action` opens up all the files I need to add a screen-action to Zellij, with helpful reminders as their titles*).

I have thoughts on where this should go, but I would very much like to hear if others have done different things here.

*because writing a layout is easier than refactoring ;)
